### PR TITLE
Validate api key parameter has credentials type

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/42267

## Description
add a validation to verify API token is in credential type (9) and not in encrypted type (4)
For more info see https://xsoar.pan.dev/docs/integrations/yaml-file#configuration

## Screenshots
Screen Shot 2021-10-21 at 12 37 24

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 * Fixed an issue in **validate** in repositories which are not in GitHub or GitLab
 * Added a validation that verifies that readme image absolute links do not contain the working branch name.
 * Added a validation to ensure reputation command's default argument is set as an array input.
+* Added a validation to verify **API token** parameter in integration configuration is in credential type.
```
Comment (amshamah419) [inline CHANGELOG.md:7]:
```suggestion
* Added a validation to verify that the**API token** parameter for an integration configuration is a known credential type.
```

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -435,6 +436,11 @@ def wrong_required_value(param_name):
     def wrong_required_type(param_name):
         return 'The type field of the {} parameter should be 8'.format(param_name)
 
+    @staticmethod
+    @error_code_decorator
+    def api_token_is_not_in_credential_type(param_name):
+        return f'The type field of the {param_name} parameter should be 9'
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/errors.py:442]:
```suggestion
        return f'In order to allow fetching the {param_name} from an external vault, the type of the {param_name} parameter should be changed from 'Encrypted' (type 4), to 'Credentials' (type 9)'. For more details check the convention for credentials - https://xsoar.pan.dev/docs/integrations/code-conventions#credentials
```

File: CHANGELOG.md (hunk 1 above)
Comment (reutshal) [reply CHANGELOG.md:8]: Please except the changes

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 3]
```diff
@@ -116,7 +119,7 @@ def is_valid_file(self, validate_rn: bool = True, skip_test_conf: bool = False,
                 validate_rn (bool): Whether to validate release notes (changelog) or not.
                 skip_test_conf (bool): If true then will skip test playbook configuration validation
                 check_is_unskipped (bool): Whether to check if the integration is unskipped.
-                conf_file (dict):
+                conf_json_data (dict):
```
Comment (reutshal) [inline demisto_sdk/commands/common/hook_validations/integration.py:122]: Please add a description

**Review by reutshal — CHANGES_REQUESTED** [review]
Nice, please see my comments

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 4]
```diff
@@ -116,7 +119,7 @@ def is_valid_file(self, validate_rn: bool = True, skip_test_conf: bool = False,
                 validate_rn (bool): Whether to validate release notes (changelog) or not.
                 skip_test_conf (bool): If true then will skip test playbook configuration validation
                 check_is_unskipped (bool): Whether to check if the integration is unskipped.
-                conf_file (dict):
+                conf_json_data (dict) The conf.json file data:
```
Comment (reutshal) [inline demisto_sdk/commands/common/hook_validations/integration.py:122]:
```suggestion
                conf_json_data (dict): The conf.json file data.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,7 @@
 * Fixed an issue where the **update-release-notes** command crashes with `--help` flag.
 * Removed a validation that DBotRole should be set for automation that requires elevated permissions to the `XSOAR-linter` in the **lint** command.
 * Fixed an issue in **Validate** command where playbooks conditional tasks were mishandeled.
+* Added a validation to ensure that the type for **API token** (and similar) parameters are configured correctly as a `credential` type in the integration configuration YML
```
Comment (reutshal) [inline CHANGELOG.md:5]:
```suggestion
* Added a validation to ensure that the type for **API token** (and similar) parameters are configured correctly as a `credential` type in the integration configuration YML.
```

**Review by reutshal — APPROVED** [review]

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**